### PR TITLE
Week 4 homework

### PR DESCRIPTION
**Pull request by ChKONS**

### CODE REVIEW

File: foundation-week-4/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+## GIT WORKFLOW FUNDAMENTALS
```
Comment (CHA) [inline foundation-week-4/README.md:1]: ✅ Great answers

File: foundation-week-4/t3q1.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+from unittest import TestCase, main
+from t2q1 import authenticate, withdrawal
+
+class HomeworkTests(TestCase):
```
Comment (CHA) [inline foundation-week-4/t3q1.py:4]: ⚠️ (-2) Great work on the unit tests. However, that would perfect if you could add the edge cases where user tries to input more than 3 times incorrect password.

File: foundation-week-4/t2q1.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#Task 1
+tries = 3
+account_balance= 100
+pin_code = 4567
+
+def authenticate(pin_code, user_pin_code):
```
Comment (CHA) [inline foundation-week-4/t2q1.py:6]: ✅ Impressive work. I like the structure of the code and the approach to making the `authenticate` and `withdrawl` methods small and testable. They are pure functions.
Improvement: Try to make custom exception when user tries to withdraw more than the available balance. Instead of `raise Exception`, you can raise your business specific exception i.e. `raise InsufficientFundError`

**Comment by CHA** [comment]
Excellent work! You managed to put-it-all together all the concepts from exception handling to unit tests.
I like the way you divide your function into small and testable onces.

Grade: 98
